src/controllers/csi/metadata/correctness.go

```diff
@@ -49,14 +49,18 @@ func (checker *CorrectnessChecker) CorrectCSI(ctx context.Context) error {
 
 // Removes volume entries if their pod is no longer exists
 func (checker *CorrectnessChecker) removeVolumesForMissingPods(ctx context.Context) error {
+	if checker.apiReader == nil {
```

Comment:
Why not moving this check to CorrectCSI() and check it only once for both removeVolumes and removeDynakubes ?

Reply:
is cleaner IMO, keeps each check independent